day10/C++/day10AllPermutations.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+* @author : YashMeh
+* @date   : 28/1/2019
+*/
+#include<bits/stdc++.h>
+using namespace std;
+//Using STL
+void permute(string s)
+{
+   sort(s.begin(),s.end());
+   do
+   {
+      cout<<s<<endl;
+   } while (next_permutation(s.begin(),s.end()));
+}
+//Using recursion
+void permute(string smallPart,int n,string firstChar)
+{
+   if(n==1)
+   {
+      cout<<firstChar+smallPart<<endl;
+      return;
+   }
+   for(int i=0;i<n;i++)
+   {
+      permute(smallPart.substr(1),n-1,firstChar+smallPart[0]);
+      rotate(smallPart.begin(),smallPart.begin()+1,smallPart.end());
+   }
+}
+main()
```

Comment:
```suggestion
void main()
```
or you can use `int` as a type specifier to make the function declaration as `int main()` in which case it is recommended to add `return 0` as the last statement.